src/ir_passes/await.ml

```diff
@@ -48,7 +48,7 @@ let ( -@- ) k exp2 =
     varE v -*- exp2
   | MetaCont (typ0, k) ->
     match exp2.it with
-    | VarE v -> k (var v (typ exp2))
+    | VarE (Const, v) -> k (var v (typ exp2))
```

Comment:
Hmmm, this could have been a real bug. But I think this is a very fringe case and the time when a mutable variable is recorded is not really specified.

Reply:
Interesting. Definitely better to introduce the let when is mutable by falling through. I wonder if  the case that v is mutable ever arose?

Reply:
I doubt it very much as the `v` carries the result of the previous computation, and CPS never allocates mutable bindings.